GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -0,0 +1,309 @@
+            this.processWrapper = new ProcessLauncher();
+            this.Output = Console.Out;
+            this.upgrader = new ProductUpgrader(ProcessHelper.GetCurrentProcessVersion(), this.tracer);
+        }
+
+        [Option(
+            "confirm",
+            Default = false,
+            Required = false,
+            HelpText = "Pass in this flag to actually install the newest release")]
+        public bool Confirmed { get; set; }
+
+        public override string EnlistmentRootPathParameter { get; set; }
+
+        protected override string VerbName
+        {
+            get { return UpgradeVerbName; }
+        }
+        
+        public override void Execute()
+        {
+            ReturnCode exitCode = ReturnCode.Success;
+            if (!this.TryRunProductUpgrade())
+            {
+                exitCode = ReturnCode.GenericError;
+                this.ReportErrorAndExit(this.tracer, exitCode, string.Empty);
+            }
+        }
+        
+        private bool TryRunProductUpgrade()
+        {
+            string errorOutputFormat = Environment.NewLine + "ERROR: {0}";
+            string error = null;
+            Version newestVersion = null;
+            bool isInstallable = false;
+
+            if (this.upgrader.IsNoneRing())
+            {
+                string message = "Upgrade ring set to None. No upgrade check was performed.";
+                this.tracer.RelatedInfo($"{nameof(this.TryRunProductUpgrade)}: {message}");
+                this.ReportInfoToConsole(message);
+                return true;
+            }
+
+            if (!this.TryRunUpgradeChecks(out newestVersion, out isInstallable, out error))
+            {
+                this.Output.WriteLine(errorOutputFormat, error);
+                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: Upgrade checks failed. {error}");
+                return false;
+            }
+
+            if (newestVersion == null)
+            {
+                this.ReportInfoToConsole($"Great news, you're all caught up on upgrades in the {this.upgrader.Ring} ring!");
+                return true;
+            }
+
+            this.ReportInfoToConsole("New GVFS version available: {0}", newestVersion.ToString());
+
+            if (!isInstallable)
+            {
+                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: {error}");
+                this.Output.WriteLine(errorOutputFormat, error);
+                return false;
+            }
+
+            if (!this.Confirmed)
+            {
+                this.ReportInfoToConsole("Run `gvfs upgrade --confirm` to install it");
+                return true;
+            }
+
+            if (!this.TryRunInstaller(out error))
+            {
+                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: Could not launch installer. {error}");
+                this.Output.WriteLine(errorOutputFormat, "Could not launch installer. " + error);

```

Comment:
Here you say "installer" but in the spinner inside `TryRunInstaller` you say "upgrade tool". Please be consistent.